### PR TITLE
fix missing datasource error message

### PR DESCRIPTION
**Pull request by timifasubaa**
When a datasource is deleted, slices show a cryptic error message in image 1. 
This PR fixes it so it shows a more intelligible error message as in image 2. 

@john-bodley @michellethomas 
screen shot 2018-05-16 at 12 23 05 pm

screen shot 2018-05-16 at 12 30 15 pm

### CODE REVIEW

**Comment by john-bodley** [comment]
Should we cascade deleted datasource and delete the corresponding slice(s)?

File: superset/views/core.py [hunk 1]
```diff
@@ -1229,7 +1229,10 @@ def datasource_info(datasource_id, datasource_type, form_data):
         datasource = form_data.get('datasource', '')
         if '__' in datasource:
             datasource_id, datasource_type = datasource.split('__')
-        datasource_id = int(datasource_id)
+            # The case where the datasource has been deleted
+            datasource_id = None if datasource_id == 'None' else datasource_id
```
Comment (mistercrunch) [inline superset/views/core.py]: I don't understand how `datasource_id == 'None'` comes into being in the first place. Can we catch the issue upstream?

**Comment by timifasubaa** [comment]
@john-bodley In the case where I found the issue, the user wasn't aware the underlying datasource was missing.

File: superset/views/core.py (hunk 1 above)
Comment (timifasubaa) [reply superset/views/core.py]: True. I was also confused at the beginning. `datasource` evaluates to 'None__Table', so splitting that gets `'None'`. I'll investigate upstream.

**Comment by timifasubaa** [comment]
@mistercrunch Upon furthher investigation, I think the current approach is best for the following reaosns 
1. datasource_info is the first place the datasource is checked from the form data. And it's the first place

2. The method intentionally does not fail because the error is handled downstream (https://github.com/apache/incubator-superset/blob/master/superset/viz.py#L60), where the intelligible error message (Viz is missing a datasource) is displayed.

**Comment by mistercrunch** [comment]
Can we change the error message to something like "The data source associated with this chart no longer exists"?

Also, should we raise right then (asap) instead of waiting for raising in `viz.py`

**Comment by mistercrunch** [comment]
About cascading, I think that would be the desired behavior as long as there's clarity on what is happening. There should be a confirmation screen with all the associated objects that are going to be deleted, and the person performing the operation should have write/delete access to all these objects.

**Comment by timifasubaa** [comment]
Agree with raising early. Done that. 

I agree with the cascading idea of forcing users to delete all other dependent slices before deleting the slice but I'll push it to a future PR.

File: superset/views/core.py [hunk 2]
```diff
@@ -1229,6 +1229,12 @@ def datasource_info(datasource_id, datasource_type, form_data):
         datasource = form_data.get('datasource', '')
         if '__' in datasource:
             datasource_id, datasource_type = datasource.split('__')
+            # The case where the datasource has been deleted
+            datasource_id = None if datasource_id == 'None' else datasource_id
+
+        if not datasource_id:
```
Comment (john-bodley) [inline superset/views/core.py]: Why not simply, 
```
if datasource_id == 'None':
    ...
```

**Review by john-bodley — APPROVED** [review]

**Pull request merged**